rcl/src/rcl/subscription.c

```diff
@@ -193,6 +193,15 @@ rcl_subscription_init(
   goto cleanup;
 fail:
   if (subscription->impl) {
+    if (subscription->impl->rmw_handle) {
+      if (rmw_destroy_subscription(
+          rcl_node_get_rmw_handle(node),
+          subscription->impl->rmw_handle) != RMW_RET_OK)
+      {
+        RCL_SET_ERROR_MSG(rmw_get_error_string().str);
```

Comment:
I'm pretty sure anytime we `goto fail`, an error is already set. This means we will always be overwriting the error.
Assuming the user probably cares more about the original error, I'd rather not overwrite the error here. @wjwwood @clalancette thoughts?

In any case, it's probably worth adding a TODO to handle this error cascade, referring to https://github.com/ros2/rcl/issues/740.

Reply:
I think you're right @jacobperron, usually we just print to stderr if there's an error while cleaning up.

Alternatively you could build a new error message which contains both, like:

> Encountered error 'SECOND ERROR' while handling error 'FIRST ERROR'.

But then again you might have an error building that new error message, so you might just say `failed to destroy subscription during error handling in rcl_subscription_init()` or something similar.

Reply:
I think this is what was being described above (in this issue) though.

Reply:
+1 for just printing the rmw error string here. Same in publisher case.

Reply:
This still is an issue for me. I'd prefer to see it print to stderr here, maybe with `RCUTILS_SAFE_FWRITE_TO_STDERR`:

https://github.com/ros2/rcutils/blob/060a68a9ef092f1ed909967ff0611a2bac5f9f18/include/rcutils/error_handling.h#L46